### ISSUE TITLE
CSV export should export names in "last, first" format

### ISSUE THREAD

**Issue by fgmart-zz** [NatTuck/bottlenose-classic#124]
As it works now, hard to alphabetize when you bring into your own spreadsheet for processing.

**Comment by marksherman** [comment]
I spent about a half hour this morning figuring out how to make excel parse it out into separate names, then re-cat them in the opposite order.

**Comment by NatTuck** [comment]
Any opposition to simply splitting first and last into two columns?

**Comment by marksherman** [comment]
I think two different fields for first and last name is a great solution. Many students have names where either their first or last is more than one word, so breaking on whitespace is unreliable. Two fields just gets it right.

**Comment by NatTuck** [comment]
Names are a mess. There are actually no general rules about name structure, and it's impossible to do "Last, First" ordering correctly - since there's no such thing as first and last names in general. Consider the proper name "Sitting Bull" - it certainly isn't correctly alphabetized as "Bull, Sitting". And it doesn't divide into a given name and a surname.

I initially designed Bottlenose to do names pedantically correctly - a free text field with no structure that users can put whatever they want in. Alphabetization then is just sorting on that field, which results in ordering by first name.

Unfortunately, that conflicts with the convention that the existing computer systems use of "Sur, Given". Which is a bad data model for anything but western European names. You'll note that every class has one Chinese student who's name on the final is backwards from their name in the class roster - and there's no consistency about which one is backwards. Half the time the roster is wrong, half the time the roster is right and the student writes their name per Chinese convention ("Sur Given").

The current Bottlenose model is a compromise that A.) always allows students to enter and see their names correctly and B.) mostly sorts names in the same order as class rosters. Unless we import class rosters directly, we'll _always_ have some mismatches.

At Northeastern, I have reliable email addresses in my class rosters, so I can sort on that to sync with Bottlenose output even if students enter nonsense for names.

In conclusion, I'm not 100% sure what the right answer is. Using the current splitting heuristic (last token = surname) and putting the surname in a second column in the export doesn't seem too bad.

**Comment by fgmart-zz** [comment]
Sounds good to me.

**Comment by marksherman** [comment]
There is an existing system to interface against, which is currently the
cause of the problems, and may provide one avenue for a solution. The
two-field mechanism is already employed in the university systems, flaws
and all. Students have already made a decision on how to interpret their
names into this system. All in all, two fields may reduce the classes of
mismatch errors, if not reduce them in total number.

My real idea is to suggest something else as the primary identifier,
possibly a number field. I use a grading spreadsheet anyway, so matching id
numbers once at the beginning of the semester would then make all my data
align up until the very end. To do reporting, i just sort my table by
whatever the university gave me.

**Comment by NatTuck** [comment]
I added a last name field to the thing.
